crux-bench/time-series-bench/src/crux/bench/utils.clj

```diff
@@ -0,0 +1,55 @@
+(ns crux.bench.utils
+  (:require [clojure.data.json :as json]
+            [crux.kafka.embedded :as ek]
+            [crux.io :as cio]
+            [clojure.java.io :as io]
+            [crux.api :as api]
+            [clojure.string :as string]
+            [clojure.java.shell :as shell]))
+
+(def commit-hash (string/trim (:out (shell/sh "git" "rev-parse" "HEAD"))))
+(def crux-version (when-let [pom-file (io/resource "META-INF/maven/juxt/crux-core/pom.properties")]
+                    (with-open [in (io/reader pom-file)]
+                      (let [{:strs [version
+                                    revision]}
+                            (cio/load-properties in)]
+                        version))))
+
+
+(defn output [mp]
+  (println (json/write-str (merge mp
+                                  {:crux-commit commit-hash
+                                   :crux-version crux-version}))))
+
+(defn bench [ingest queries bench-ns]
```

Comment:
subjective: this `bench` function is assuming quite a lot about the structure of a benchmark test - any way we can split this into its component parts so that each part is doing one thing?

maybe:
* `with-node`, starts kafka, passes node to callback, cleans up after
* `with-bench-timing`, takes callback, times fn, outputs `(merge result-of-fn {:ingest-time-ms ...})` in JSON format
* `queries` goes - it's 'just' part of the function passed to `with-node` (albeit that the queries will likely call `with-bench-timing`)

obv if you ask 5 developers to structure code into functions, you'll get 5 different suggestions - thoughts, though?

Reply:
I'd certainly agree splitting out node startup + having with-bench-timing, unsure on queries

Reply:
in hindsight, this is best left as-is til we get a few more usage examples